cylc/flow/scripts/psutil.py

```diff
@@ -48,10 +48,17 @@ def get_option_parser():
 
 def _psutil(metrics_json):
     metrics = parse_dirty_json(metrics_json)
-    methods = [
-        getattr(psutil, key[0])
-        for key in metrics
-    ]
+
+    try:
+        methods = [
+            getattr(psutil, key[0])
+            for key in metrics
+        ]
+    except AttributeError as exc:
+        # error obtaining interfaces from psutil e.g:
+        # * requesting a method which does not exist
+        print(exc, file=sys.stderr)
+        sys.exit(2)
```

Comment:
The `2` return code is used to help differentiate errors in the ranking expression from unexpected errors.